packages/job-manager/src/batch/batch.ts

```diff
@@ -26,6 +28,61 @@ export class Batch {
         @inject(Logger) private readonly logger: Logger,
     ) {}
 
+    public async getPoolTaskProcessingRatio(): Promise<number> {
+        const resultVariableName = 'taskProcessingRatioResult';
+        const poolTaskProcessingRatioFormula = `
+        // The script will calculate the tasks processing ratio returned in $taskProcessingRatio variable as
+        // double value and will return -1 when there are no active tasks or sampling data is insufficient
+
+        // the time interval for data sampling
+        $sampleTimeIntervalMinutes = 15 * TimeInterval_Minute;
```

Comment:
is this formula something we will be changing at runtime? if not shouldnt this be part of arm template / deployment?
also, autoscaling will break if you go beyond 50 nodes / downscale to 0 & then scale up.

Reply:
We will not change this at runtime and this does not belong to the deployment. Also it has no connection with 50's node limit issue. It uses batch stat data to calculate average pool load. Note this is not auto scale implementation.

---
In reply to: [305949078](https://github.com/microsoft/accessibility-insights-service/pull/221#discussion_r305949078) [](ancestors = 305949078)